### ISSUE TITLE
PHP 7 cause PayPal gateway has rejected request. Timeout processing request (#10001: Internal Error).

### ISSUE THREAD

**Issue by zhipeng001** [magento/magento2#2878]
I have tested on my linode host and my local MAMP.  When you go to checkout it will cause the error "PayPal gateway has rejected request. Timeout processing request (#10001: Internal Error)."

**Comment by ifekaj** [comment]
I have the same problem. This is what I have in my exception.log:

```
[2016-01-25 16:40:18] main.CRITICAL: Magento\Framework\Exception\LocalizedException: PayPal gateway has rejected request. Timeout processing request (#10001: Internal Error). in /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php:99
Stack trace:
#0 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(89): Magento\Framework\ObjectManager\Factory\AbstractFactory->createObject('Magento\\Framewo...', Array)
#1 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Framewo...', Array)
#2 /var/www/html/var/generation/Magento/Framework/Exception/LocalizedExceptionFactory.php(43): Magento\Framework\ObjectManager\ObjectManager->create('Magento\\Framewo...', Array)
#3 /var/www/html/vendor/magento/module-paypal/Model/Api/Nvp.php(1295): Magento\Framework\Exception\LocalizedExceptionFactory->create(Array)
#4 /var/www/html/vendor/magento/module-paypal/Model/Api/Nvp.php(1241): Magento\Paypal\Model\Api\Nvp->_handleCallErrors(Array)
#5 /var/www/html/vendor/magento/module-paypal/Model/Api/Nvp.php(1026): Magento\Paypal\Model\Api\Nvp->call('getPalDetails', Array)
#6 /var/www/html/vendor/magento/module-paypal/Model/Express/Checkout.php(385): Magento\Paypal\Model\Api\Nvp->callGetPalDetails()
#7 /var/www/html/vendor/magento/module-paypal/Block/Express/Shortcut.php(181): Magento\Paypal\Model\Express\Checkout->getCheckoutShortcutImageUrl()
#8 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(651): Magento\Paypal\Block\Express\Shortcut->_beforeToHtml()
#9 /var/www/html/vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#10 /var/www/html/vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('wpsexpress\\shor...')
#11 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('wpsexpress\\shor...')
#12 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('wpsexpress\\shor...')
#13 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('wpsexpress\\shor...', true)
#14 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(496): Magento\Framework\View\Layout\Interceptor->renderElement('wpsexpress\\shor...', true)
#15 /var/www/html/vendor/magento/module-catalog/Block/ShortcutButtons.php(94): Magento\Framework\View\Element\AbstractBlock->getChildHtml()
#16 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(652): Magento\Catalog\Block\ShortcutButtons->_toHtml()
#17 /var/www/html/vendor/magento/module-checkout/Block/Cart.php(210): Magento\Framework\View\Element\AbstractBlock->toHtml()
#18 /var/www/html/vendor/magento/module-checkout/view/frontend/templates/cart/methods.phtml(17): Magento\Checkout\Block\Cart->getMethodHtml('checkout.cart.s...')
#19 /var/www/html/vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/var/www/html/v...')
#20 /var/www/html/vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Checkout\Block\Cart), '/var/www/html/v...', Array)
#21 /var/www/html/vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template->fetchView('/var/www/html/v...')
#22 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(652): Magento\Framework\View\Element\Template->_toHtml()
#23 /var/www/html/vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#24 /var/www/html/vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('checkout.cart.m...')
#25 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.m...')
#26 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.m...')
#27 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('checkout.cart.m...', true)
#28 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.m...')
#29 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('cart.summary')
#30 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('cart.summary')
#31 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('cart.summary')
#32 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('cart.summary', true)
#33 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('cart.summary')
#34 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('checkout.cart.c...')
#35 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.c...')
#36 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.c...')
#37 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('checkout.cart.c...', true)
#38 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.c...')
#39 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('checkout.cart.i...')
#40 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart.i...')
#41 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart.i...')
#42 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('checkout.cart.i...', true)
#43 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(492): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart.i...', true)
#44 /var/www/html/vendor/magento/module-checkout/view/frontend/templates/cart.phtml(15): Magento\Framework\View\Element\AbstractBlock->getChildHtml('with-items')
#45 /var/www/html/vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/var/www/html/v...')
#46 /var/www/html/vendor/magento/framework/View/Element/Template.php(255): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Checkout\Block\Cart), '/var/www/html/v...', Array)
#47 /var/www/html/vendor/magento/framework/View/Element/Template.php(279): Magento\Framework\View\Element\Template->fetchView('/var/www/html/v...')
#48 /var/www/html/vendor/magento/framework/View/Element/AbstractBlock.php(652): Magento\Framework\View\Element\Template->_toHtml()
#49 /var/www/html/vendor/magento/framework/View/Layout.php(542): Magento\Framework\View\Element\AbstractBlock->toHtml()
#50 /var/www/html/vendor/magento/framework/View/Layout.php(518): Magento\Framework\View\Layout->_renderBlock('checkout.cart')
#51 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('checkout.cart')
#52 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('checkout.cart')
#53 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('checkout.cart', true)
#54 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('checkout.cart')
#55 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('content')
#56 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('content')
#57 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('content')
#58 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('content', true)
#59 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('content')
#60 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('main')
#61 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('main')
#62 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main')
#63 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('main', true)
#64 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('main')
#65 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('columns')
#66 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('columns')
#67 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('columns')
#68 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('columns', true)
#69 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('columns')
#70 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('main.content')
#71 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('main.content')
#72 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main.content')
#73 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('main.content', true)
#74 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('main.content')
#75 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('page.wrapper')
#76 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#77 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#78 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('page.wrapper', true)
#79 /var/www/html/vendor/magento/framework/View/Layout.php(569): Magento\Framework\View\Layout\Interceptor->renderElement('page.wrapper')
#80 /var/www/html/vendor/magento/framework/View/Layout.php(520): Magento\Framework\View\Layout->_renderContainer('root')
#81 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(206): Magento\Framework\View\Layout->renderNonCachedElement('root')
#82 /var/www/html/vendor/magento/framework/View/Layout.php(472): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#83 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(193): Magento\Framework\View\Layout->renderElement('root', true)
#84 /var/www/html/vendor/magento/framework/View/Layout.php(938): Magento\Framework\View\Layout\Interceptor->renderElement('root')
#85 [internal function]: Magento\Framework\View\Layout->getOutput()
#86 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(144): call_user_func_array(Array, Array)
#87 /var/www/html/var/generation/Magento/Framework/View/Layout/Interceptor.php(494): Magento\Framework\View\Layout\Interceptor->___callPlugins('getOutput', Array, Array)
#88 /var/www/html/vendor/magento/framework/View/Result/Page.php(241): Magento\Framework\View\Layout\Interceptor->getOutput()
#89 /var/www/html/vendor/magento/framework/View/Result/Layout.php(162): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#90 [internal function]: Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#91 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(74): call_user_func_array(Array, Array)
#92 /var/www/html/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\View\Result\Page\Interceptor->___callParent('renderResult', Array)
#93 /var/www/html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-varnish-...')
#94 /var/www/html/vendor/magento/module-page-cache/Model/Controller/Result/VarnishPlugin.php(74): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#95 [internal function]: Magento\PageCache\Model\Controller\Result\VarnishPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#96 /var/www/html/vendor/magento/framework/Interception/Chain/Chain.php(67): call_user_func_array(Array, Array)
#97 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(136): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'renderResult', Object(Magento\Framework\View\Result\Page\Interceptor), Array, 'result-builtin-...')
#98 /var/www/html/vendor/magento/module-page-cache/Model/Controller/Result/BuiltinPlugin.php(67): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Response\Http\Interceptor))
#99 [internal function]: Magento\PageCache\Model\Controller\Result\BuiltinPlugin->aroundRenderResult(Object(Magento\Framework\View\Result\Page\Interceptor), Object(Closure), Object(Magento\Framework\App\Response\Http\Interceptor))
#100 /var/www/html/vendor/magento/framework/Interception/Interceptor.php(140): call_user_func_array(Array, Array)
#101 /var/www/html/var/generation/Magento/Framework/View/Result/Page/Interceptor.php(130): Magento\Framework\View\Result\Page\Interceptor->___callPlugins('renderResult', Array, Array)
#102 /var/www/html/vendor/magento/framework/App/Http.php(119): Magento\Framework\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#103 /var/www/html/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#104 /var/www/html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#105 {main} [] []
```

This is my debug.log:

```
[2016-01-25 17:36:39] main.DEBUG: array (
  'url' => 'https://api-3t.sandbox.paypal.com/nvp',
  'SetExpressCheckout' => 
  array (
    'PAYMENTACTION' => 'Sale',
    'AMT' => '79.00',
    'CURRENCYCODE' => 'SGD',
    'RETURNURL' => 'https://www.xxxxxxxxxxxxxxxx/paypal/express/return/',
    'CANCELURL' => 'https://www.xxxxxxxxxxxxxxxxx/paypal/express/cancel/',
    'INVNUM' => '000000001',
    'SOLUTIONTYPE' => 'Sole',
    'GIROPAYCANCELURL' => 'https://www.xxxxxxxxxxxxxxxxx/paypal/express/cancel/',
    'GIROPAYSUCCESSURL' => 'https://www.xxxxxxxxxxxxxxxx/checkout/onepage/success/',
    'BANKTXNPENDINGURL' => 'https://www.xxxxxxxxxxxxx/checkout/onepage/success/',
    'SHIPPINGAMT' => '0.00',
    'ITEMAMT' => '79.00',
    'TAXAMT' => '0.00',
    'L_NUMBER0' => NULL,
    'L_NAME0' => 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
    'L_QTY0' => 1,
    'L_AMT0' => '79.00',
    'BUSINESS' => NULL,
    'NOTETEXT' => NULL,
    'EMAIL' => 'xxxxxxxxxxxxxxxx',
    'FIRSTNAME' => 'xxxxxxxxx',
    'LASTNAME' => 'xxxxxxxx',
    'MIDDLENAME' => NULL,
    'SALUTATION' => NULL,
    'SUFFIX' => NULL,
    'COUNTRYCODE' => 'SG',
    'STATE' => NULL,
    'CITY' => 'Singapore',
    'STREET' => 'xxxxxxxxxxxxxxx',
    'ZIP' => 'xxxxxxxxxxxxx',
    'PHONENUM' => 'xxxxxxxxxx',
    'SHIPTOCOUNTRYCODE' => 'SG',
    'SHIPTOSTATE' => NULL,
    'SHIPTOCITY' => 'Singapore',
    'SHIPTOSTREET' => 'xxxxxxxxxxxxxx',
    'SHIPTOZIP' => 'xxxxxxxxxxxxx',
    'SHIPTOPHONENUM' => 'xxxxxxxxx',
    'SHIPTOSTREET2' => '',
    'STREET2' => '',
    'SHIPTONAME' => 'xxxxxxxx',
    'ADDROVERRIDE' => 1,
    'METHOD' => 'SetExpressCheckout',
    'VERSION' => '72.0',
    'USER' => '****',
    'PWD' => '****',
    'SIGNATURE' => '****',
    'BUTTONSOURCE' => 'Magento_Cart_Community',
  ),
  'response' => 
  array (
    'TIMESTAMP' => '2016-01-25T17:36:38Z',
    'CORRELATIONID' => 'xxxxxxxxxx',
    'ACK' => 'Failure',
    'L_ERRORCODE0' => '10001',
    'L_SHORTMESSAGE0' => 'Internal Error',
    'L_LONGMESSAGE0' => 'Timeout processing request',
  ),
) {"is_exception":false} []
[2016-01-25 17:36:43] main.DEBUG: array (
  'url' => 'https://api-3t.sandbox.paypal.com/nvp',
  'getPalDetails' => 
  array (
    'METHOD' => 'getPalDetails',
    'VERSION' => '72.0',
    'USER' => '****',
    'PWD' => '****',
    'SIGNATURE' => '****',
    'BUTTONSOURCE' => 'Magento_Cart_Community',
  ),
  'response' => 
  array (
    'TIMESTAMP' => '2016-01-25T17:36:43Z',
    'CORRELATIONID' => 'xxxxxxxxxxxxxxx',
    'ACK' => 'Failure',
    'L_ERRORCODE0' => '10001',
    'L_SHORTMESSAGE0' => 'Internal Error',
    'L_LONGMESSAGE0' => 'Timeout processing request',
  ),
) {"is_exception":false} []
```

Everything works fine when I downgrade to PHP 5.6

**Comment by alena-marchenko** [comment]
created internal ticket MAGETWO-48387

**Comment by digitalLumberjack** [comment]
:+1: same here on paypal and on braintree with php 7 magento 2.0.2

**Comment by alena-marchenko** [comment]
The issue has been resolved and delivered to develop branch.

**Comment by ortega-io** [comment]
Is there a workaround or patch to fix this until a new version is released? Could someone please point me to the files involved on the issue and the modifications needed to solve it?

**Comment by ortega-io** [comment]
@alena-marchenko  Could you please point me to the commit that fix the issue? I really need a patch for this. I will be in debt.

**Comment by dkvashninbay** [comment]
@ortega-io https://github.com/magento/magento2/commit/32ca5c97304a1bd84cfbee7cec3d57c9307da9a6

**Comment by ortega-io** [comment]
@dkvashninbay  Thank you so much! that solves it.

**Comment by sinoning** [comment]
i am facing the same problem, but my mangeto 2 no have lib/internal/Magento this folder, what to do?

**Comment by KrystynaKabannyk** [comment]
Hello @zhipeng001, this issue has been fixed in the 2.0.7 patch release and I'm closing it. Thanks for reporting the issue! If you have additional questions or proposals you are welcome to create another one.
